src/main/scala/strawman/collection/immutable/HashMap.scala

```diff
@@ -104,6 +100,12 @@ object HashMap extends MapFactory[HashMap] {
 
   def empty[K, V]: HashMap[K, V] = EmptyHashMap.asInstanceOf[HashMap[K, V]]
 
+  def fromIterable[K, V](it: collection.Iterable[(K, V)]): HashMap[K, V] =
+    it match {
+      case hm: HashMap[K, V] => hm
+      case _ => empty ++ it
```

Comment:
I didn’t measure but builders for immutable collections do nothing more than creating an empty collection and then adding the elements one by one. So, that seems equivalent. I would expect not using builders could only improve performance because that’s one abstraction less.